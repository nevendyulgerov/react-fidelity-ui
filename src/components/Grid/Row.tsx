import * as React from 'react';
import * as PropTypes from 'prop-types';
import classNames from 'classnames';
import { isNonEmptyStr } from '../../utils';

export type RowProps = {
  children: React.ReactNode,
  className?: string | null,
  [key: string]: any
};

const Row = (props: RowProps) => {
  const { children, className, ...restProps } = props;
  const componentClassName: string = classNames({
    row: true,
    // @ts-ignore
    [className]: isNonEmptyStr(className)
  });

  return (
    <div
      {...restProps}
      className={componentClassName}
    >
      {children}
    </div>
  );
};

Row.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node)
  ]).isRequired,
  className: PropTypes.string
};

Row.defaultProps = {
  className: null
};

export default Row;
