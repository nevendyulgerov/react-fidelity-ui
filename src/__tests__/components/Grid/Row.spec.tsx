import * as React from 'react';
import { render } from '@testing-library/react';
import Row from '../../../components/Grid/Row';

describe('Components / Grid / Row', () => {
  it('matches snapshot', () => {
    const { baseElement } = render(<Row>Row</Row>);
    expect(baseElement).toMatchSnapshot();
  });
});
