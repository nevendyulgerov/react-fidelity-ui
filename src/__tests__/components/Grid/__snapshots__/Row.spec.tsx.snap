// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / Grid / Row matches snapshot 1`] = `
<body>
  <div>
    <div
      class="row"
    >
      Row
    </div>
  </div>
</body>
`;
